### PR TITLE
feat(cni): expose main functions of cni and installer

### PR DESCRIPTION
**Pull request by slonka**
Expose main functions of cni and installer to be reusable

Signed-off-by: slonka <slonka@users.noreply.github.com>

### Summary

Expose main functions of cni and installer to be reusable

### Issues resolved

Build error reported on slack

### Documentation

- [x] No new docs needed  ~~Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)~~

### Testing

- ~~[ ] Unit tests~~
- ~~[ ] E2E tests~~
- [x] Manual testing on Universal
- ~~[ ] Manual testing on Kubernetes~~

### Backwards compatibility

- ~~[ ] Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.~~
- ~~[ ] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)~~

### CODE REVIEW

File: app/cni/cmd/install/main.go [hunk 1]
```diff
@@ -1,241 +1,9 @@
...
-const (
-	kumaCniBinaryPath  = "/opt/cni/bin/kuma-cni"
-	primaryBinDir      = "/host/opt/cni/bin"
-	secondaryBinDir    = "/host/secondary-bin-dir"
-	serviceAccountPath = "/var/run/secrets/kubernetes.io/serviceaccount"
-)
-
-var (
-	log = core.NewLoggerWithRotation(2, "/tmp/install-cni.log", 100, 0, 0).WithName("install-cni")
+	"github.com/kumahq/kuma/app/cni/pkg/install"
```
Comment (slonka) [inline app/cni/cmd/install/main.go:4]: Note for reviewers: this changed into a simple call, not sure why git detected it as change, this should be a new file + a move.

File: app/cni/cmd/kuma-cni/main.go [hunk 2]
```diff
@@ -1,252 +1,9 @@
...
-	"github.com/kumahq/kuma/pkg/core"
-	"github.com/kumahq/kuma/pkg/plugins/runtime/k8s/metadata"
-	"github.com/kumahq/kuma/pkg/plugins/runtime/k8s/util"
-	kuma_version "github.com/kumahq/kuma/pkg/version"
-)
-
-const (
-	podRetrievalMaxRetries = 30
-	podRetrievalInterval   = 1 * time.Second
+	"github.com/kumahq/kuma/app/cni/pkg/cni"
```
Comment (slonka) [inline app/cni/cmd/kuma-cni/main.go:4]: As the previous comment, this is a new file + a file move.

File: app/cni/pkg/cni/main.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
+package cni
```
Comment (slonka) [inline app/cni/pkg/cni/main.go:1]: old file, just package changed

File: app/cni/pkg/cni/main.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+func cmdDel(args *skel.CmdArgs) error {
+	return nil
+}
+
+func main() {
+	skel.PluginMain(cmdAdd, cmdCheck, cmdDel, version.All, fmt.Sprintf("kuma-cni %v", kuma_version.Build.Version))
+}
+
+func Run() {
+	main()
```
Comment (jakubdyszkiewicz) [inline app/cni/pkg/cni/main.go:255]: placing main in package that is not `main` is a bit awkward. I'd inline the main here and call this file `run.go`

File: app/cni/pkg/install/main.go [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+	}
+
+	if err := runLoop(installerConfig); err != nil {
+		log.Error(err, "checking installation failed - exiting")
+		os.Exit(1)
+	}
+}
+
+func Run() {
+	main()
```
Comment (jakubdyszkiewicz) [inline app/cni/pkg/install/main.go:229]: same here

**Review by jakubdyszkiewicz — APPROVED** [review]

File: app/cni/pkg/cni/main.go (hunk 4 above)
Comment (slonka) [reply app/cni/pkg/cni/main.go]: inlined

**Pull request merged**